### PR TITLE
[google_maps_flutter_web] Adds missing MapOptions parameters.

### PR DESCRIPTION
**Pull request by Franreno**
As the [linked issue](https://github.com/flutter/flutter/issues/104111) states, there are some `MapOptions` parameters that can't be modified on the `MapConfiguration` when using `google_maps_flutter_web` package. 

The parameters are:
- `mapTypeControl`: Enables the user to change mapTypes.
- `fullscreenControl`: Enables the user make the map fullscreen.
- `streetViewControl`: Enables the user to use the streetview functionality.

There parameters where previously set to false by default at the `MapConfiguration` class. With this Pull Request, these parameters could be setted to true or false.

As the previous version handles these options as false by default, I opted to keep them as false if the user has not setted them to `true`.

Using this approach and creating a simple example app that uses this new implementation, we can see that these new controllers can be used on the `google_maps_flutter_web` package.

https://github.com/flutter/packages/assets/67326251/fcdb2e67-f4d2-4780-b562-e704ecfab400

Fixes [#104111](https://github.com/flutter/flutter/issues/104111)

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by ditman** [comment]
(Discussed on chat with @stuartmorgan and we decided to go with the PRs as they are, and then maybe do a more thorough refactor of per-platform settings, for example "liteMode" doesn't make sense in the web or iOS either)

**Comment by ditman** [comment]
@Franreno there's some automated "checks" failing, please work on the branch until all automated checks pass, otherwise we can't merge it. Currently, there's the following warnings/errors being reported:

```
Analyzing google_maps_flutter_web...

  error - example/integration_test/google_maps_controller_test.dart:411:13 - The named parameter 'mapTypeControl' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'mapTypeControl'. - undefined_named_parameter
  error - example/integration_test/google_maps_controller_test.dart:412:13 - The named parameter 'fullscreenControl' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'fullscreenControl'. - undefined_named_parameter
  error - example/integration_test/google_maps_controller_test.dart:413:13 - The named parameter 'streetViewControl' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'streetViewControl'. - undefined_named_parameter
  error - lib/src/convert.dart:77:42 - The getter 'mapTypeControl' isn't defined for the type 'MapConfiguration'. Try importing the library that defines 'mapTypeControl', correcting the name to the name of an existing getter, or defining a getter or field named 'mapTypeControl'. - undefined_getter
  error - lib/src/convert.dart:78:45 - The getter 'fullscreenControl' isn't defined for the type 'MapConfiguration'. Try importing the library that defines 'fullscreenControl', correcting the name to the name of an existing getter, or defining a getter or field named 'fullscreenControl'. - undefined_getter
  error - lib/src/convert.dart:79:45 - The getter 'streetViewControl' isn't defined for the type 'MapConfiguration'. Try importing the library that defines 'streetViewControl', correcting the name to the name of an existing getter, or defining a getter or field named 'streetViewControl'. - undefined_getter
   info - example/integration_test/google_maps_controller_test.dart:410:33 - The value of the argument is redundant because it matches the default value. Try removing the argument. - avoid_redundant_argument_values

7 issues found.
```

You can find more on the ["View more" link](https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20analyze%20master/1085/overview) of each [check](https://github.com/flutter/packages/pull/4553/checks?check_run_id=15272666485) (search for the stdout of the failing step).


In this case, the analyzer is complaining about missing types because this PR isn't setup with the correct dependencies. Read instructions on how this PR needs to be configured here:

* https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-federated-plugins

**Comment by ditman** [comment]
(Moving this to WIP until this is ready to review!)

**Comment by FranrenoLV** [comment]
Hello @ditman, I really appreciate your suggestions and thank you for helping me!

> (Discussed on chat with @stuartmorgan and we decided to go with the PRs as they are, and then maybe do a more thorough refactor of per-platform settings, for example "liteMode" doesn't make sense in the web or iOS either)

As your comment states, this PR would be valid if the tests are passing? If so, only the web google maps would receive these new parameters?

I'm currently working to get the tests fixed!
As always, thank you for your help.

**Comment by Franreno** [comment]
Hello @ditman, I really appreciate your suggestions and thank you for helping me!

> (Discussed on chat with @stuartmorgan and we decided to go with the PRs as they are, and then maybe do a more thorough refactor of per-platform settings, for example "liteMode" doesn't make sense in the web or iOS either)

As your comment states, this PR would be valid if the tests are passing? If so, only the web google maps would receive these new parameters?

I'm currently working to get the tests fixed!
As always, thank you for your help.

**Comment by Franreno** [comment]
@ditman I think the `version_check` check is failing because I haven't updated the changelogs for the other packages that were included in the dependancies override. If I shall update them, please, let me know.

**Comment by Franreno** [comment]
> Packages that don't actually need dependency overrides should just have the dependency overrides reverted.

I have reverted the other packages that didn't needed the dependancies overrides. 

The only checks failing are the `submit-queue` and the `Linux repo_checks`. I think the repo_checks failing is supposed to happen because I'm making changes to the platform_interface package. But I don't know why the `submit-queue` is failing.

**Comment by stuartmorgan** [comment]
> But I don't know why the `submit-queue` is failing.

It's unrelated to the PR; see https://github.com/flutter/flutter/wiki/Understanding-Packages-tests#specific-tests

**Comment by Franreno** [comment]
Oh okay! Thanks!

**Comment by Franreno** [comment]
Would this PR be valid for review now? I there is anything else that I need to do let me know.

**Comment by ditman** [comment]
Yes, this is ready for review now @Franreno, thanks for keeping the CI happy!

**Review by reidbaker — APPROVED** [review]
Android +1

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart [hunk 1]
```diff
@@ -107,6 +110,16 @@ class MapConfiguration {
   /// True if 3D building display should be enabled.
   final bool? buildingsEnabled;
 
+  /// True if the control to toggle between map types should be displayed.
+  final bool? mapTypeControl;
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:114]: These should all end with `Enabled`, like the existing comparable properties.

File: packages/google_maps_flutter/google_maps_flutter_android/example/pubspec.yaml [hunk 2]
```diff
@@ -32,3 +32,8 @@ flutter:
   uses-material-design: true
   assets:
     - assets/
+
+# FOR TESTING AND INITIAL REVIEW ONLY. DO NOT MERGE.
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_android/example/pubspec.yaml:36]: Please revert the changes to `google_maps_flutter_android` and `google_maps_flutter_ios`.

File: packages/google_maps_flutter/google_maps_flutter/example/pubspec.yaml
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/example/pubspec.yaml:1]: Why aren't there any actual changes in this package? Isn't the intent of the PR to allow clients of `google_maps_flutter` to configure these options?

File: packages/google_maps_flutter/google_maps_flutter_web/example/integration_test/google_maps_controller_test.dart [hunk 3]
```diff
@@ -403,6 +403,30 @@ void main() {
           expect(capturedOptions!.tilt, 0);
         });
 
+        testWidgets(
+            'translates mapTypeControl, fullscreenControl, streetViewControl configurations',
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_web/example/integration_test/google_maps_controller_test.dart:407]: Three unrelated things should not be tested in the same test.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 2.4.1
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md:1]: New functionality is a minor update, not a bugfix update.

File: packages/google_maps_flutter/google_maps_flutter_web/pubspec.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@ name: google_maps_flutter_web
 description: Web platform implementation of google_maps_flutter
 repository: https://github.com/flutter/packages/tree/main/packages/google_maps_flutter/google_maps_flutter_web
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+maps%22
-version: 0.5.2
+version: 0.5.2+1
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_web/pubspec.yaml:5]: Same here; this should be `0.5.3`.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by reidbaker** [comment]
@Franreno is this still something you are planning to work on?

**Comment by stuartmorgan** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**